src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -819,5 +832,27 @@ private static string GenerateGlobalExtendedAttributeName(int globalExtendedAttr
 
             return result;
         }
+
+        // Returns the text's utf16 length truncated at the specified max length.
+        private static int GetUtf16TruncatedTextLength(ReadOnlySpan<char> text, int maxLength)
+        {
+            int utf8Length = 0;
+            int utf16TruncatedLength = 0;
+
+            foreach (Rune rune in text.EnumerateRunes())
+            {
+                utf8Length += rune.Utf8SequenceLength;
+                if (utf8Length <= maxLength)
+                {
+                    utf16TruncatedLength += rune.Utf16SequenceLength;
```

Comment:
I don't understand the logic here.  For each rune, you're adding both how many bytes it consumes as Utf8 and how many chars it consumes? Why?

Reply:
Oh! I see, you're trying to figure out how many of the chars can fit in the max length when encoded.  Ok.

Reply:
To know how many chars are able to fit encoded in a byte buffer of size `maxLength`.